compiler-plugin-tests/src/test/resources/ballerina_sources/validator_tests/valid_service_17/service.bal

```diff
@@ -22,16 +22,46 @@ public type Person record {
     Person[] friends;
 };
 
-service /graphql on new graphql:Listener(4000) {
+public type Book readonly & record {|
+    string name;
+    string author;
+|};
 
+service /graphql on new graphql:Listener(4000) {
     isolated resource function get name(Person & readonly person) returns string {
         return person.name;
     }
+
+    isolated resource function get name(readonly & Person[] person) returns string {
+        return person.name;
+    }
 }
 
 service /graphql on new graphql:Listener(4000) {
-
     isolated resource function get profile() returns Person {
         return {name: "Walter White", age: 52, friends: []};
     }
 }
+
+service on new graphql:Listener(4000) {
+    isolated resource function get profile() returns Person & readonly {
+        return {name: "Walter White", age: 52, friends: []}.cloneReadOnly();
+    }
+
+    isolated resource function get profiles() returns readonly & Person[] {
+        return [{name: "Walter White", age: 52, friends: []}].cloneReadOnly();
+    }
+}
+
+service on new graphql:Listener(4000) {
+    resource function get book() returns Book {
+        return {
+            name: "The Fabric of the Cosmos",
+            author: "Brian Green"
+        };
+    }
+
+    resource function get books() returns Book[] {
+        return [{name: "Magic of Reality", author: "Richard Dawkins"}];
+    }
+}
```

Comment:
Have addressed the possible name collision ?

Reply:
What is the name collision you are expecting?

Reply:
I was thinking something like the below. Wouldn't it be a problem ?

```ballerina
type Teacher record {|
    string id;
    record {|
        string first;
        string last;
    |} name;
|};

type Student record {|
    string id;
    record {|
        string first;
        string last;
        string middle;
    |} name;
|};
```

Reply:
This will give a compilation error.

Reply:
I see. We expect the user to resolver the conflict. What is the error message in that case ?

Reply:
This is an expected error message: https://github.com/ThisaruGuruge/module-ballerina-graphql/blob/9b811b39227e0cb246b3d8b52d344a4cee497595/compiler-plugin-tests/src/test/java/io/ballerina/stdlib/graphql/compiler/ServiceValidationTest.java#L859

Reply:
This is also mentioned in the spec now. (Spec is updated in this PR).

Reply:
Sorry, my bad. I was under the impression that we were generating names to solve the problem. Because it is common to have anonymous records Ballerina. 

@ThisaruGuruge and I had a call on this and decided to go ahead with this fix. Later, we will explore the possibility of improving this behavior by generating names for anonymous records. 

As a minor improvement we decided to improve the error message to help user locate the problematic anonymous record.